email_scraper/src/main/java/EmailScraper.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public String getHelp() {
+        return "java -jar EmailScraper-all.jar [my_url]";
+    }
+
+    // find all urls that start with domain and add them to the HashSet
+    private void getPageLinks() {
+        // list of web pages to be examined
+        queue.add("http://" + domain);
+
+        while (!queue.isEmpty()) {
+            String v = queue.remove();
+
+            // skip https to avoid jsoup's exception: "java.net.UnknownHostException: https"
+            // potential solution: https://stackoverflow.com/questions/11392614/unknownhostexception-while-accessing-https-url-using-jsoup
+            if (v.startsWith("https://")) {
+                continue;
+            }
+//            System.out.println(v);
+
+            try {
+                final Document doc = Jsoup.connect(v).get();
+                getEmails(doc);
+                ++link_count;
+
+                // visit links only if total number of visited links is not greater than NUMBER_OF_LINKS
+                if (link_count > NUMBER_OF_LINKS) {
+                    break;
+                }
+
+                final Elements links2go = doc.select("a[href]");
+                for (Element link : links2go) {
+                    final String link_str = link.attr("abs:href");
+
+                    // visit links only if not being visited
+                    if (links.add(link_str)) {
+//                        System.out.println(getDomainName(link_str));
+
+                        // visit links only if not sub-domain
+                        if (getDomainName(link_str).equals(domain)) {
+                            queue.add(link_str);
+//                            System.out.println(link_str);
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                // handle "org.jsoup.HttpStatusException: HTTP error fetching URL. Status=404, ..."
+//                System.err.println("For '" + v + "': " + e.getMessage());
+            }
+        }
+    }
+
+    public String getDomainName(String url) {
+        try {
+            URI uri = new URI(url);
+            String domain = uri.getHost();
+            if (domain != null) {
+                return domain.startsWith("www.") ? domain : "www." + domain;
+            } else {
+                return "";
+            }
+        } catch (URISyntaxException e) {
+//            System.err.println("For '" + url + "': " + e.getMessage());
+            return "";
+        }
+    }
+
+    private void getEmails(Document doc) {
+        Pattern p = Pattern.compile("[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+");
+        Matcher matcher = p.matcher(doc.text());
+        while (matcher.find()) {
+            String email_str = matcher.group();
+            if (email_str.endsWith(".")) {
+                email_str = email_str.substring(0, email_str.length() - 1);
+            }
+            emails.add(email_str);
+//            System.out.println(email_str);
```

Comment:
I actually think you probably want this print in place, rather than just printing everything out at the end. Then you can put your cap at 10k or 100k or whatever, and have the script print emails as it's finding them, rather than waiting an hour or whatever for the script to run before seeing anything.

Reply:
Good point. I have changed this part to immediately print the emails once found.